app/src/components/inputs/inputTagsSearch/inputTagsSearch.jsx

```diff
@@ -0,0 +1,152 @@
+import Select, { AsyncCreatable, Async, Creatable } from 'react-select';
+import React, { Component } from 'react';
+import { FormattedMessage } from 'react-intl';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import { ScrollWrapper } from 'components/main/scrollWrapper';
+import { fetch } from 'common/utils';
+import styles from './inputTagsSearch.scss';
+
+const cx = classNames.bind(styles);
+
+export class InputTagsSearch extends Component {
+  static propTypes = {
+    uri: PropTypes.string,
+    options: PropTypes.array,
+    values: PropTypes.array,
+    focusPlaceholder: PropTypes.string,
+    loadingPlaceholder: PropTypes.string,
+    nothingFound: PropTypes.string,
+    isCreatable: PropTypes.bool,
+    isAsync: PropTypes.bool,
+    multi: PropTypes.bool,
+    removeSelected: PropTypes.bool,
+    selectItem: PropTypes.func,
+    makeOptions: PropTypes.func,
+    isValidNewOption: PropTypes.func,
+    validation: PropTypes.func,
+    minLength: PropTypes.number,
+    showNewLabel: PropTypes.bool,
+    dynamicSearchPromptText: PropTypes.bool,
+  };
+  static defaultProps = {
+    uri: '',
+    options: [],
+    values: [],
+    focusPlaceholder: '',
+    loadingPlaceholder: '',
+    nothingFound: '',
+    isCreatable: false,
+    isAsync: false,
+    multi: true,
+    removeSelected: false,
+    makeOptions: () => {},
+    isValidNewOption: () => {},
+    validation: () => {},
+    selectItem: () => {},
+    minLength: 1,
+    showNewLabel: false,
+    dynamicSearchPromptText: false,
+  };
+  static renderItems(params) {
+    return (
+      <ScrollWrapper autoHeight autoHeightMax={200}>
+        {Select.defaultProps.menuRenderer(params)}
+      </ScrollWrapper>
+    );
+  }
+  static optionRenderer(option, selectValue) {
+    return (<div
+      className={cx('select2-item')}
+      key={option.value}
+      onClick={() => { selectValue(option); }}
+    >
+      <span>{option.label}</span>
+    </div>);
+  }
+  state = {
+    value: this.props.values,
+    searchPromptText: this.props.nothingFound,
+  };
+  onChange(value) {
+    this.setState({ value });
+    this.props.selectItem(value);
+  }
+  onInputChange(input) {
+    const diff = this.props.minLength - input.length;
+    if (this.props.dynamicSearchPromptText && this.props.minLength && diff > 0) {
+      const dynamicSearchPromptText = <FormattedMessage
+        id={'InputTagsSearch.dynamicSearchPromptText'}
+        defaultMessage={'Please enter {length} or more characters'}
+        values={{ length: diff }}
+      />;
+      this.setState({ searchPromptText: dynamicSearchPromptText });
+    } else {
+      this.setState({ searchPromptText: this.props.nothingFound });
+    }
+    return input;
+  }
+  getItems(input, uri) {
```

Comment:
```
getItems = (input) => {
//...
}
```

Defining class method like this guarantees that a function will not lose its' context when using in JSX.